src/applications/vre/definitions/profileAddress.js

```diff
@@ -0,0 +1,265 @@
+ * @param {function} callback - slots into ui:required for the necessary fields
+ *
+ * Conventions:
+ * 1. formDataPath - path to entire address property in formData, after accounting for potential array nesting. Derived from the path parameter
+ * 2. get(formDataPath, formData) - returns the address property, often destructured with const {country, isMilitary} = get(formDataPath, formData)
+ *
+ * Examples:
+ * 1. Path to Address nested in array - childrenToAdd[INDEX].childAddressInfo.address
+ */
+
+export const addressUiSchema = (path, checkBoxTitle, callback) => {
+  // Used when addresses are nested in an array and need to be assessible.
+  const insertArrayIndex = (key, index) => key.replace('[INDEX]', `[${index}]`);
+  // the address is nested in an array, handle index
+  const getPath = (pathToData, index) =>
+    typeof index === 'number'
+      ? insertArrayIndex(pathToData, index)
+      : pathToData;
+
+  return (function returnAddressUI() {
+    return {
+      isMilitary: {
+        'ui:title': checkBoxTitle,
+      },
+      'view:livesOnMilitaryBaseInfo': {
+        'ui:description': MilitaryBaseInfo,
+      },
+      country: {
+        'ui:required': callback,
+        'ui:title': 'Country',
+        'ui:options': {
+          updateSchema: (formData, schema, uiSchema, index) => {
+            const formDataPath = getPath(path, index);
+            const countryUI = uiSchema;
+            const addressFormData = get(formDataPath, formData);
+            const { isMilitary } = addressFormData;
+            // if isMilitary === true, auto select United States and disable the field
+            if (isMilitary) {
+              countryUI['ui:disabled'] = true;
+              addressFormData.country = USA.value;
+              return {
+                enum: [USA.value],
+                enumNames: [USA.label],
+                default: USA.value,
+              };
+            }
+            // default to regular country select dropdown
+            countryUI['ui:disabled'] = false;
+            return {
+              type: 'string',
+              enum: countries.map(country => country.value),
+              enumNames: countries.map(country => country.label),
+            };
+          },
+        },
+      },
+      addressLine1: {
+        'ui:required': callback,
+        'ui:title': 'Street',
+        'ui:errorMessages': {
+          required: 'Street address is required',
+          pattern: 'Street address must be under 100 characters',
+        },
+      },
+      addressLine2: {
+        'ui:title': 'Line 2',
+      },
+      addressLine3: {
+        'ui:title': 'Line 3',
+      },
+      city: {
+        'ui:required': callback,
+        'ui:errorMessages': {
+          required: 'City is required',
+          pattern: 'City must be under 100 characters',
+        },
+        'ui:options': {
+          replaceSchema: (formData, schema, uiSchema, index) => {
```

Comment:
Same as on the `updateSchema` callback, I'd recommend writing a JSDoc for this to help future maintainers understand the _why_ behind this.